lib/puma/launcher.rb

```diff
@@ -311,6 +313,32 @@ def prune_bundler
       end
     end
 
+    #
+    # Puma's systemd integration allows Puma to inform systemd:
+    #  1. when it has successfully started
+    #  2. when it is starting shutdown
+    #  3. periodically for a liveness check with a watchdog thread
+    #
+
+    def integrate_with_systemd
+      return unless ENV["SD_NOTIFY"]
```

Comment:
Why would you want this non-standard check? Isn't the presence of `NOTIFY_SOCKET` sufficient? If needed, you can also check if the system was booted with systemd. See https://github.com/sj26/puma-plugin-systemd/blob/8490c0e3136ddd201bef93c6747e1057f0d74a03/lib/puma/plugin/systemd.rb#L159-L167